lib/let_me/builder.ex

```diff
@@ -68,8 +68,16 @@ defmodule LetMe.Builder do
     error_reason = Keyword.fetch!(opts, :error_reason)
     rule_clauses = Enum.map(rules, &permit_function_clause(&1, check_module))
 
+    typespec =
+      rules
+      |> Map.keys()
+      |> Enum.reduce(&{:|, [], [&1, &2]})
+
     quote do
+      @type actions() :: unquote(typespec)
```

Comment:
I think this should be singular.

```suggestion
      @type action() :: unquote(typespec)
```